src/main/java/org/opensearch/index/codec/customcodecs/ZstdNoDictCompressionMode.java

```diff
@@ -30,15 +32,15 @@ public class ZstdNoDictCompressionMode extends CompressionMode {
 
     /** default constructor */
     protected ZstdNoDictCompressionMode() {
-        this.compressionLevel = Lucene99CustomCodec.DEFAULT_COMPRESSION_LEVEL;
+        this.compressionLevel = DEFAULT_COMPRESSION_LEVEL;
     }
 
     /**
      * Creates a new instance with the given compression level.
      *
      * @param compressionLevel The compression level.
      */
-    protected ZstdNoDictCompressionMode(int compressionLevel) {
+    public ZstdNoDictCompressionMode(int compressionLevel) {
```

Comment:
why are we changing the access modifiers?

Reply:
This the way Java interpret the visibility rules (see please https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html). In the nutshell - `ZstdNoDictCompressionMode` is used from 2 different packages now which the `protected` modifier prevents (we may do the subclassing though)